tensorboard/webapp/notification_center/_redux/notification_center_reducers.ts

```diff
@@ -14,16 +14,38 @@ limitations under the License.
 ==============================================================================*/
 import {Action, createReducer, on} from '@ngrx/store';
 import * as actions from './notification_center_actions';
-import {NotificationState} from './notification_center_types';
+import {
+  NotificationState,
+  NOTIFICATION_LAST_READ_TIME_KEY,
+} from './notification_center_types';
 import {notificationNotes} from './notification_notes';
 
-/** @typehack */ import * as _typeHackStore from '@ngrx/store';
-
 const initialState: NotificationState = {
   notifications: notificationNotes,
+  lastReadTimestamp: window.localStorage.getItem(
+    NOTIFICATION_LAST_READ_TIME_KEY
+  )
+    ? parseInt(window.localStorage.getItem(NOTIFICATION_LAST_READ_TIME_KEY)!)
+    : 0,
 };
 
-const reducer = createReducer(initialState);
+const reducer = createReducer(
+  initialState,
+  on(
+    actions.notificationBellClicked,
+    (state: NotificationState): NotificationState => {
+      const timeNow = Date.now();
+      window.localStorage.setItem(
+        NOTIFICATION_LAST_READ_TIME_KEY,
+        timeNow.toString()
+      );
```

Comment:
Are you going to move this to a DataSource? Please add a TODO to indicate that this is not the final state.

Reply:
added